src/service/locator.cpp

```diff
@@ -550,16 +581,18 @@ locator_t::on_refresh(const std::vector<std::string>& groups) {
         return accumulator;
     });
 
-    const auto ruids = boost::accumulate(*m_routers.synchronize(), ruid_vector_t{},
+    const auto ruids = boost::accumulate(*routing.routers.synchronize(), ruid_vector_t{},
         [](ruid_vector_t result, const router_map_t::value_type& value) -> ruid_vector_t
     {
         result.push_back(value.first); return result;
     });
 
+    const auto results = routing.dump();
+
     for(auto it = ruids.begin(); it != ruids.end(); ++it) try {
```

Comment:
See #242 for more details. As I mentioned it conflicts with it, still I don't want to copy/paste changes.